dnb_direct_plus/monitoring.py

```
@@ -62,16 +109,91 @@ def add_companies_to_dnb_monitoring_registration():
         return pending_registrations.count()
 
 
+def _parse_timestamp_from_file(file_name):
+    """Parse the date from the filename, which is in the format: {RegistrationNumber_{timestamp}_[...].{ext}"}"""
+
+    raw_time = os.path.basename(file_name.split('_')[1])
+
+    timestamp = datetime.datetime.strptime(raw_time, '%Y%m%d%H%M%S')
```

Are we guaranteed that the filename will always be in that format? If there's any deviation for any reason (e.g. a rogue file or something) then this will bomb out. Maybe put a `try` / `catch` round this to catch the `ValueError`?

---

Good question - they haven't given a guarantee.  I think I would rather it fail hard and fast if th e file is in a different format. In fact, it should ideally kill the import job dead, but I'm not sure this happens currently. This is because Company.last_updated_source_timestamp gets set to the timestamp and older timestamps won't be applied. So if a file gets skipped, potentially that update will be lost forever.